internal/indexer/indexer_test.go

```diff
@@ -336,3 +337,44 @@ func compareRange(t *testing.T, r protocol.Range, startLine, startCharacter, end
 		)
 	}
 }
+
+func TestIndexer_shouldVisitPackage(t *testing.T) {
+	w := &capturingWriter{}
+	projectRoot := getRepositoryRoot(t)
+	indexer := New(
+		"/dev/github.com/sourcegraph/lsif-go/internal/testdata",
+		projectRoot,
+		protocol.ToolInfo{Name: "lsif-go", Version: "dev"},
+		"testdata",
+		"0.0.1",
+		nil,
+		w,
+		NewPackageDataCache(),
+		OutputOptions{},
+	)
+
+	if err := indexer.loadPackages(); err != nil {
+		t.Fatal(err)
+	}
+
+	visited := map[string]bool{}
+	for _, pkg := range indexer.packages {
+		shortID := strings.Replace(pkg.ID, "github.com/sourcegraph/lsif-go/internal/testdata/internal", "…", -1)
+		if indexer.shouldVisitPackage(pkg) {
+			visited[shortID] = true
+		} else {
+			visited[shortID] = false
+		}
+	}
+	autogold.Want("visited", map[string]bool{
```

Comment:
I pulled in my [autogold](https://github.com/hexops/autogold) project here. I can imagine this being super useful in other places of this codebase - but I'm happy to do whatever here, just let me know what your preference is.

Reply:
I have no issue adopting autogold + similar tools (things here are just out of past convenience). If you want to update any tests to use snapshots feel free to add additional or replace old coverage.